### PR TITLE
Implementation for HKY85 nucleotide rate matrix

### PR DESCRIPTION
**Pull request by lsdch**

### CODE REVIEW

File: lib/rate_matrix.ml [hunk 1]
```diff
@@ -138,6 +138,30 @@ module Nucleotide = struct
...
+          else alpha_transition
+        in
+        coef *. stationary_distribution.Nucleotide.%(j)
+      ) in
+    scaled_rate_matrix stationary_distribution m
+
+  let%test "HKY85 stationary distribution" =
+    let rng = Utils.rng_of_int 420 in
+    let pi = Nucleotide.random_profile rng 10. in
+    let rate_params = Utils.random_profile rng 2 in
```
Comment (pveber) [inline lib/rate_matrix.ml:158]: rates have no reason to sum to one, they just need to be positive. Maybe it's better not to test on this particular case

**Review by pveber — CHANGES_REQUESTED** [review]
Splendid. I only have minor requests.

File: lib/rate_matrix.ml [hunk 2]
```diff
@@ -138,6 +138,30 @@ module Nucleotide = struct
         else if Nucleotide.transversion i j then 1. /. (kappa +. 2.)
         else kappa /. (kappa +. 2.)
       )
+
+  let hky85 ~equilibrium_frequencies:(stationary_distribution : Nucleotide.vector)
+      ~alpha_transition ~beta_transversion =
```
[the remark this answers is not in the document]
Comment (lsdch) [reply lib/rate_matrix.ml]: Addressed in 1870749

File: lib/rate_matrix.ml (hunk 1 above)
Comment (lsdch) [reply lib/rate_matrix.ml]: ac5fe5e uses rates drawn from uniform distribution, without constraint on the sum

Comment (pveber) [reply lib/rate_matrix.ml]: :ok_hand:

**Review by pveber — APPROVED** [review]

**Comment by pveber** [comment]
thx!

**Pull request merged**